### PR TITLE
executor: change to structured slow log for friendly JSON output

### PR DESCRIPTION
**Pull request by breeswish**
Hint: to output in JSON format, you need to use the following config:

```toml
[log]
format = "json"
```

### Sample outputs after applying this PR

JSON format:

```
{"connectionId":1,"costTime":1000142797,"file":"adapter.go","level":"info","line":274,"msg":"query","sql":"select sleep(1)","time":"2017/09/27 14:20:47.209"}
```

Normal text format:

```
2017/09/27 14:27:02.535 adapter.go:274: [info] query costTime=1.0001397s sql=select sleep(1) connectionId=1
```

@shenli PTAL

### CODE REVIEW

**Comment by coocood** [comment]
LGTM

File: executor/adapter.go [hunk 1]
```diff
@@ -263,10 +263,15 @@ func (a *statement) logSlowQuery() {
...
 	}
 	connID := a.ctx.GetSessionVars().ConnectionID
+	logEntry := log.WithFields(log.Fields{
+		"connectionId": connID,
+		"costTime":     costTime,
+		"sql":          sql,
+	})
 	if costTime < time.Duration(cfg.Log.SlowThreshold)*time.Millisecond {
-		log.Debugf("[%d][TIME_QUERY] %v %s", connID, costTime, sql)
+		logEntry.Debugf("query")
```
Comment (zimulala) [inline executor/adapter.go]: Why change `TIME_QUERY` to `query` ?

Comment (breeswish) [reply executor/adapter.go]: I found it actually logs all queries, not just for slow queries.

File: executor/adapter.go [hunk 2]
```diff
@@ -263,10 +263,15 @@ func (a *statement) logSlowQuery() {
...
 	} else {
-		log.Warnf("[%d][TIME_QUERY] %v %s", connID, costTime, sql)
+		logEntry.Infof("query")
```
Comment (shenli) [inline executor/adapter.go]: can we change it to slow-query?

Comment (shenli) [reply executor/adapter.go]: We should use an unique string to mark this log. The use could grep it from log file.

Comment (zimulala) [reply executor/adapter.go]: @breeswish

Comment (breeswish) [reply executor/adapter.go]: @shenli done

**Comment by shenli** [comment]
LGTM

File: executor/adapter.go [hunk 3]
```diff
@@ -263,10 +263,15 @@ func (a *statement) logSlowQuery() {
...
+		"connectionId": connID,
+		"costTime":     costTime,
+		"sql":          sql,
+	})
 	if costTime < time.Duration(cfg.Log.SlowThreshold)*time.Millisecond {
-		log.Debugf("[%d][TIME_QUERY] %v %s", connID, costTime, sql)
+		logEntry.Debugf("query")
 	} else {
-		log.Warnf("[%d][TIME_QUERY] %v %s", connID, costTime, sql)
+		logEntry.Infof("slow-query")
```
Comment (winkyao) [inline executor/adapter.go]: use Warnf?

**Comment by shenli** [comment]
@winkyao PTAL

**Pull request merged**